main.tf

```diff
@@ -149,6 +149,28 @@ resource "azurerm_kubernetes_cluster" "main" {
       }
     }
   }
+  dynamic "maintenance_window" {
+    for_each = var.maintenance_window != null ? ["maintenance_window"] : []
+
+    content {
+      dynamic "allowed" {
+        for_each = var.maintenance_window.allowed
+
+        content {
+          day   = allowed.value.day
+          hours = allowed.value.hours
+        }
+      }
+      dynamic "not_allowed" {
+        for_each = var.maintenance_window.not_allowed
+
+        content {
```

Comment:
Thanks @zioproto for the review. I tend not to add this `precondition` since the [provider](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/containers/kubernetes_cluster_resource.go#L376-L398) didn't limit this. Maybe the caller just generate this `not_allowed` list by code, or read it from their CMDB, maybe they just don't care whether if the parameters contain a past time or not.